job.go

```diff
@@ -207,3 +207,18 @@ func (j *Job) String() string {
 		"{project=%s params=%s group=%s id=%s}",
 		j.Project, j.Params, j.Group, j.ID[:7])
 }
+
+// State determines the job's current state by using it's path in the filesystem.
+func (j *Job) State(path, project, id string) (string, error) {
+	pPath := filepath.Join(path, project, "pending", id)
+	rPath := filepath.Join(path, project, "ready", id)
```

Comment:
Can't we use `j.PendingBuildPath` and `j.ReadyBuildPath`?

Reply:
I don't think we can as the job has been initialized with an empty struct 
```go
j := &Job{}
state, err := j.GetState(s.cfg.BuildPath, project, id)
```

Probably we can introduce a `GetJob` function that would find a job, WDYT?

Reply:
Didn't notice that. Jobs should not be initialized this way, we should only use `NewJob()`.